### PR TITLE
Update Domain Registration Result screen UI

### PR DESCRIPTION
**Pull request by renanferrari**
This PR updates the UI from the Domain Registration Result screen:

|Light Mode|Dark Mode|
|-----|-----|
|domains-registration-result-light|domains-registration-result-dark|

The following is out of scope for this PR:
- Cart creation and checkout error handling.
- Updating the domains dashboard after purchase.
- A few UI improvements to the Suggestions screen.
- We're still going to add a progress bar after tapping the "Choose domain" button on the Suggestions screen.

### To test

**Setup: Enable feature flag**

1. On the Main screen, tap the Me button.
1. On the Me screen, go to App Settings > Debug settings.
1. On the Debug Settings screen, scroll down to the "Features in development" section and tap `SiteDomainsFeatureConfig` to enable the Domains feature flag.

**Scenario 1: Redeem domain with credits**

1. Open the app and select a site that has domain credits.
1. On the Main screen, tap the "Register Domain" item.
1. Go through the entire flow.
1. Notice the new Domain Registration Result screen.

**Scenario 2: Purchase domain**

1. Open the app and select a site that has **no** domain credits.
1. On the Main screen, tap the "Domains" item under the "Configuration" section.
1. On the Domains Dashboard screen, tap the "Search for a domain" button.
1. Go through the entire flow.
1. Notice the new Domain Registration Result screen.

**Extra**

Try to press back instead of the "Done" button in the result screen and make sure you go back to the Domains Dashboard.

## Regression Notes
1. Potential unintended areas of impact
Existing domain flows.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests.

3. What automated tests I added (or what prevented me from doing so)
Updated existing unit tests.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/WindowExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import android.view.View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+import android.view.View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+import android.view.View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
+import android.view.View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+import android.view.Window
+import androidx.core.content.ContextCompat
+
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/util/WindowExtensions.kt:11]: Nice 🙌

**Review by ravishanker — APPROVED** [review]
LGTM 👍

**Comment by kyleaparker** [comment]
This is looking good! I have a couple of updates:

On dark mode, I think we can style it similar to the initial app screen with a blue button and slightly lighter background behind the button: 
Screen Shot 2021-10-13 at 7 50 49 PM

If it's difficult to change the colour behind the button, it's ok to keep it as is but I think the button looks better in blue. 

I was also thinking we may want to increase the heading size to match the heading size on the initial app screen, and if possible prevent the orphan at the end of the sentence.

**Comment by renanferrari** [comment]
Thanks for taking a look at this @kyleaparker! This is how the screens look like after I applied your suggestions:

|Light Mode|Dark Mode|
|-----|-----|
|domains-registration-result-light-updated|domains-registration-result-dark-updated|

I'm going to merge this now, but if there's anything else we should change, please let me know and I'll address it on a future PR.

**Pull request merged**